### PR TITLE
Migrate design system to Material You

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## :scroll: Description

Migrates APS to use [Material You](https://m3.material.io) themes and components wherever possible, cleaning up hard-coded colors and themes as well as performing various other cleanups in the process.

We also enable support for [dynamic colors](https://m3.material.io/styles/color/dynamic-color/overview) on API 31 so Android 12 users can have a consistent UI with the rest of their system and Google apps.

## :bulb: Motivation and Context

Material You is the next iteration of Material Design and the library ecosystem around it has reached sufficient maturity for us to be able to migrate to it.

## :green_heart: How did you test it?

Visual verification of all screens.

## :pencil: Checklist

- [x] I formatted the code `./gradlew spotlessApply`
- [x] I reviewed submitted code
- [x] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

## :crystal_ball: Next steps

- `Dialog`s used by our preference library still rely on AppCompat and hence do not respect the new theming/styling changes, being tracked [here](https://github.com/Maxr1998/ModernAndroidPreferences/issues/16).
- Manrope might not be the best font face for the app, I'm open to suggestions.

## :camera_flash: Screenshots / GIFs

<details>
<summary>Screenshots (light + dark Material You dynamic colors)</summary>

screenshot-20211103-164239
screenshot-20211103-164230
screenshot-20211103-164214
screenshot-20211103-164207
screenshot-20211103-164132
screenshot-20211103-164124
screenshot-20211103-164117
screenshot-20211103-164109
screenshot-20211103-163529
screenshot-20211103-163523
screenshot-20211103-163517
screenshot-20211103-163509
screenshot-20211103-163447
screenshot-20211103-163440
screenshot-20211103-163432
screenshot-20211103-163423
screenshot-20211103-163410
screenshot-20211103-163402
screenshot-20211103-163355
screenshot-20211103-163348
screenshot-20211103-163245
screenshot-20211103-163236
screenshot-20211103-163226
screenshot-20211103-163211

</details>

<details>
<summary>Screenshots (new color palette, no dynamic colors)</summary>

screenshot-20211103-164430
screenshot-20211103-164425
screenshot-20211103-164421
screenshot-20211103-164415
screenshot-20211103-164404
screenshot-20211103-164357

</details>

### CODE REVIEW

**Comment by Skrilltrax** [comment]
Damn, the app looks lit af :fire:

File: app/src/main/res/values/themes_material3.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    <item name="colorOnErrorContainer">@color/md_theme_light_onErrorContainer</item>
+    <item name="android:colorBackground">@color/md_theme_light_background</item>
+    <item name="colorOnBackground">@color/md_theme_light_onBackground</item>
+    <item name="colorSurface">@color/md_theme_light_surface</item>
+    <item name="colorOnSurface">@color/md_theme_light_onSurface</item>
+    <item name="colorSurfaceVariant">@color/md_theme_light_surfaceVariant</item>
+    <item name="colorOnSurfaceVariant">@color/md_theme_light_onSurfaceVariant</item>
+    <item name="colorOutline">@color/md_theme_light_outline</item>
+    <item name="colorOnSurfaceInverse">@color/md_theme_light_inverseOnSurface</item>
+    <item name="colorSurfaceInverse">@color/md_theme_light_inverseSurface</item>
```
Comment (Skrilltrax) [inline app/src/main/res/values/themes_material3.xml:71]: nit: can we rename these to `onSurfaceInverse` and `surfaceInverse` to follow the pattern?

File: app/src/main/res/values/colors_material3.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  <color name="md_theme_dark_onErrorContainer">#FFDAD4</color>
+  <color name="md_theme_dark_background">#1A1C1E</color>
+  <color name="md_theme_dark_onBackground">#E1E2E5</color>
+  <color name="md_theme_dark_surface">#1A1C1E</color>
+  <color name="md_theme_dark_onSurface">#E1E2E5</color>
+  <color name="md_theme_dark_surfaceVariant">#41474D</color>
+  <color name="md_theme_dark_onSurfaceVariant">#C1C7CE</color>
+  <color name="md_theme_dark_outline">#8B9197</color>
+  <color name="md_theme_dark_inverseOnSurface">#1A1C1E</color>
+  <color name="md_theme_dark_inverseSurface">#E1E2E5</color>
```
Comment (Skrilltrax) [inline app/src/main/res/values/colors_material3.xml:57]: nit: can we rename these to `onSurfaceInverse` and `surfaceInverse` everywhere?

File: app/src/main/res/values/colors_material3.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  <color name="md_theme_light_onError">#FFFFFF</color>
+  <color name="md_theme_light_onErrorContainer">#410001</color>
+  <color name="md_theme_light_background">#FCFCFF</color>
+  <color name="md_theme_light_onBackground">#1A1C1E</color>
+  <color name="md_theme_light_surface">#FCFCFF</color>
+  <color name="md_theme_light_onSurface">#1A1C1E</color>
+  <color name="md_theme_light_surfaceVariant">#DDE3EA</color>
+  <color name="md_theme_light_onSurfaceVariant">#41474D</color>
+  <color name="md_theme_light_outline">#72787E</color>
+  <color name="md_theme_light_inverseOnSurface">#F0F0F3</color>
```
Comment (Skrilltrax) [inline app/src/main/res/values/colors_material3.xml:30]: nit: can we rename these to onSurfaceInverse and surfaceInverse to follow the pattern?

File: app/src/main/res/values-night/themes_material3.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    <item name="colorOnErrorContainer">@color/md_theme_dark_onErrorContainer</item>
+    <item name="android:colorBackground">@color/md_theme_dark_background</item>
+    <item name="colorOnBackground">@color/md_theme_dark_onBackground</item>
+    <item name="colorSurface">@color/md_theme_dark_surface</item>
+    <item name="colorOnSurface">@color/md_theme_dark_onSurface</item>
+    <item name="colorSurfaceVariant">@color/md_theme_dark_surfaceVariant</item>
+    <item name="colorOnSurfaceVariant">@color/md_theme_dark_onSurfaceVariant</item>
+    <item name="colorOutline">@color/md_theme_dark_outline</item>
+    <item name="colorOnSurfaceInverse">@color/md_theme_dark_inverseOnSurface</item>
+    <item name="colorSurfaceInverse">@color/md_theme_dark_inverseSurface</item>
```
Comment (Skrilltrax) [inline app/src/main/res/values-night/themes_material3.xml:33]: nit: can we rename these to onSurfaceInverse and surfaceInverse to follow the pattern?

File: app/src/main/res/values/colors_material3.xml (hunk 2 above)
Comment (msfjarvis) [reply app/src/main/res/values/colors_material3.xml:57]: These tokens were imported as-is from the new Material Theme Builder so I'm opposed to making nomenclature changes that diverge from upstream recommendations.

**Review by Skrilltrax — APPROVED** [review]
Looks pretty good to me :+1:

File: app/src/main/res/values/colors_material3.xml (hunk 2 above)
Comment (Skrilltrax) [reply app/src/main/res/values/colors_material3.xml:57]: Oh cool, weird that they don't follow the pattern like the other tokens

**Comment by msfjarvis** [comment]
ModernAndroidPreferences has been updated to allow clients to provide their own [dialog builder factory](https://github.com/Maxr1998/ModernAndroidPreferences/commit/3ff76d7afc6d33cf6847c577004d4e9f1d74489a), once the next release happens we'll have consistent M3 dialogs everywhere.

**Pull request merged**